### PR TITLE
Fixes #75: Replace libimobiledevice with pymobiledevice3

### LINKED ISSUE #75: Switch to pymobiledevice3
It would seem that https://github.com/doronz88/pymobiledevice3/ would solve a lot of our iOS related problems. It already bundles `usbmuxd` and `lockdownd` for us in a PyPI package, which is amazing and would make #71 unnecessary, since we already have a way to manage python dependencies in our projects. It also supports everything we currently use `libimobiledevice` for and more:
- Install and list apps on the device
- Show device information
- Port forwarding (#43)
- Collecting the network traffic with `pcapd` and attributing is to specific processes
- Install configuration profiles (#44)
- Mount developer image and use developer functions, e.g. setting a honey location or starting apps

### PR DESCRIPTION
**Pull request by zner0L**
This should be merged after https://github.com/tweaselORG/autopy/issues/5 has been fixed, as the setup of the venv will not work otherwise.

### CODE REVIEW

File: src/ios.ts [hunk 1]
```diff
@@ -151,15 +150,23 @@ export const iosApi = <RunTarget extends SupportedRunTarget<'ios'>>(
     async waitForDevice(tries = 100) {
         if (
             !(await retryCondition(
-                async () => (await execa('ideviceinfo', ['-k', 'DeviceName'], { reject: false })).exitCode === 0,
+                // Actually wait until the SpringBoard has been started and users could interact with the device.
+                async () =>
+                    (await python('pymobiledevice3', ['springboard', 'state', 'get'], { reject: false })).exitCode ===
```
Comment (baltpeter) [inline src/ios.ts:155]: This has a zero exit code even if no device is connected:

```sh
❯ pymobiledevice3 springboard state get
2023-06-02 19:01:32 ryzen pymobiledevice3.__main__[4107830] ERROR Failed to connect to usbmuxd socket. Make sure it's running.
❯ echo $?
0
```

File: src/ios.ts [hunk 2]
```diff
@@ -183,27 +190,16 @@ export const iosApi = <RunTarget extends SupportedRunTarget<'ios'>>(
...
     installApp: async (ipaPath) => {
-        if (process.platform === 'win32') await execa('ideviceinstaller', ['install', ipaPath]);
-        else await execa('ideviceinstaller', ['--install', ipaPath]);
+        await python('pymobiledevice3', ['apps', 'install', ipaPath]);
     },
     uninstallApp: async (appId) => {
-        if (process.platform === 'win32') await execa('ideviceinstaller', ['uninstall', appId]);
-        else await execa('ideviceinstaller', ['--uninstall', appId]);
+        await python('pymobiledevice3', ['apps', 'uninstall', appId]);
     },
```
Comment (baltpeter) [inline src/ios.ts:203]: Now that we don't need the `if` anymore, you can just return the promise.

File: src/ios.ts [hunk 3]
```diff
@@ -183,27 +190,16 @@ export const iosApi = <RunTarget extends SupportedRunTarget<'ios'>>(
...
+        const { stdout } = await python('pymobiledevice3', ['apps', 'list', '--user', '--no-color']);
+        return Object.keys(JSON.parse(stdout)).includes(appId);
     },
     // We're using `libimobiledevice` instead of `cfgutil` because the latter doesn't wait for the app to be fully
     // installed before exiting.
     installApp: async (ipaPath) => {
-        if (process.platform === 'win32') await execa('ideviceinstaller', ['install', ipaPath]);
-        else await execa('ideviceinstaller', ['--install', ipaPath]);
+        await python('pymobiledevice3', ['apps', 'install', ipaPath]);
     },
```
Comment (baltpeter) [inline src/ios.ts:200]: Same here.

**Review by baltpeter — CHANGES_REQUESTED** [review]
Very cool.

File: src/ios.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zner0L) [reply src/ios.ts:155]: Again, I cannot reproduce this. Can your provide a little more information?

[the remark this answers is not in the document]
Comment (baltpeter) [reply src/ios.ts]: Not quite sure what you're looking for. I installed using `pipx install pymobiledevice3` and when I run `pymobiledevice3 springboard state get` with the iOS 15 phone connected, nothing happens and it just hangs. On iOS 16, it dumps out JSON.

File: src/ios.ts [hunk 4]
```diff
@@ -182,29 +190,13 @@ export const iosApi = <RunTarget extends SupportedRunTarget<'ios'>>(
...
-    uninstallApp: async (appId) => {
-        if (process.platform === 'win32') await execa('ideviceinstaller', ['uninstall', appId]);
-        else await execa('ideviceinstaller', ['--uninstall', appId]);
-    },
+    isAppInstalled: (appId) =>
+        python('pymobiledevice3', ['apps', 'list', '--user', '--no-color']).then(({ stdout }) =>
+            Object.keys(JSON.parse(stdout)).includes(appId)
+        ),
+    installApp: (ipaPath) => python('pymobiledevice3', ['apps', 'install', ipaPath]).then(),
+    uninstallApp: (appId) => python('pymobiledevice3', ['apps', 'uninstall', appId]).then(),
```
[the remark this answers is not in the document]
Comment (baltpeter) [reply src/ios.ts:198]: Ah, okay.

File: src/ios.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (baltpeter) [reply src/ios.ts]: [Newly jailbroken](https://github.com/tweaselORG/meta/issues/4#issuecomment-1578543235), it still works on that device. So I guess the timeout is fine as a workaround.

**Review by baltpeter — APPROVED** [review]

**Pull request merged**